cidc_api/models.py

```diff
@@ -89,6 +122,28 @@ def create(email: str, session: Session = None):
         return user
 
 
+class Permissions(CommonColumns):
+    __tablename__ = "permissions"
+
+    id = Column(Integer, primary_key=True, autoincrement=True, nullable=False)
+
+    # If user who granted this permission is deleted, this permission will be deleted.
+    # TODO: is this what we want?
+    granted_by_user = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"))
+
+    granted_to_user = Column(
+        Integer, ForeignKey("users.id", ondelete="CASCADE"), index=True
+    )
+    trial_id = Column(
+        String,
+        ForeignKey("trial_metadata.trial_id", ondelete="CASCADE"),
+        nullable=False,
+        index=True,
+    )
+    assay_type = Column(Enum(*SUPPORTED_ASSAYS, name="assays"), nullable=False)
```

Comment:
I just wonder - there was such a requirement that perms are given wrt to specific assays only? 
If not - wouldn't it be a permission granted always on all assays, so we will end up storing tens of perm entries instead of just one?

Reply:
I don't know of an explicit specification, so this was a bit speculative. Maybe making the `assay` field nullable would allow us to represent both trial-wide and trial-assay-level permissions (i.e., if empty, then the user can view all assays for that trial)? Alternatively, `assay_type` could instead be an array `assay_types`, containing all the assays a user is permitted to view.

Reply:
I see. I think it will probably be okay to keep it like it is, if we don't expect a sheer number of users. And from my understanding we don't.